constrain-lib/src/main/resources/schema/dependency-constraints-schema.json

```diff
@@ -0,0 +1,123 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "type": "object",
+  "title": "Dependency Constraints Schema.",
+  "required": [
+    "dependencyConstraints"
+  ],
+  "properties": {
+    "dependencyConstraints": {
+      "$ref": "#/$defs/dependencyConstraints"
+    }
+  },
+  "additionalProperties": false,
+  "$defs": {
+    "dependencyConstraints": {
+      "type": "array",
+      "title": "Constraints",
+      "description": "A collection of Dependency Constraints.",
+      "items": {
+        "$ref": "#/$defs/dependencyConstraint"
+      }
+    },
+    "dependencyConstraint": {
+      "type": "object",
+      "title": "Constraint",
+      "description": "A single Dependency Constraint.",
+      "required": [
+        "groupId",
+        "name",
+        "suggestedVersion",
+        "because"
+      ],
+      "properties": {
+        "groupId": {
+          "type": "string",
+          "title": "Group",
+          "description": "The group of the dependency.",
+          "examples": [
+            "com.example"
+          ]
+        },
+        "name": {
```

Comment:
I think it is common to call this `artifactId` if you call the other field `groupId`. Or, if you want to follow Gradle nomenclature then the other field would have to be named `group`.

Reply:
I agree. I kinda did this this way explicitly to trigger this conversation as I didn't know which way to go on this.

@big-guy @ljacomet which direction would you want to see this named?

Reply:
We decided to call this `group` in line with the existing Gradle API for declaring dependencies.